singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -353,12 +353,12 @@ class PTESolverBase {
       Asum += vfrac[m] * robust::ratio(press[m], temp[m]);
       rhoBsum += rho[m] * vfrac[m] * robust::ratio(sie[m], temp[m]);
     }
-    PORTABLE_REQUIRE(Tnorm > 0., "Non-positive temperature guess");
+    PORTABLE_REQUIRE(Tnorm > 0., "Non-positive Ideal PTE temperature guess");
     Asum *= uscale / Tnorm;
     rhoBsum /= Tnorm;
-    PORTABLE_REQUIRE(rhoBsum > 0., "Non-positive energy density");
+    PORTABLE_REQUIRE(rhoBsum > 0., "Non-positive Ideal PTE energy density guess");
```

Comment:
The next line is
```c++
Tideal = uscale / rhoBsum / Tnorm;
```

This would imply a negative temperature guess, so no, this code cannot admit a negative energy density. I think this is probably part of why it's not useful outside of gas mixtures.

Reply:
Ah ok. I just want to make sure this is only used for ideal gas, as tabulated EOSs definitely support negative energy.

Reply:
yep! I think this is a good reason _not_ to use this guess for tabular EOS. You could also imagine a reacting system where the reaction products is represented by a shifted ideal gas with a negative energy. I think this guess isn't really used, but I figured it might be good to put guardrails on it.

Reply:
This function is called in the `Init` for `PTESolverRhoU` which I believe is supposed to handle non-ideal EOS.

Notice that `TryIdealPTE` is commented out from `PTESolverRhoT` with the following comment:

```c++
    // Leave this in for now, but comment out because I'm not sure it's a good idea
    // TryIdealPTE(this);
    // Set the current guess for the equilibrium temperature.  Note that this is already
    // scaled.
```

Reply:
I unresolved this conversation because I was concerned that `PTESolverRhoU ` now no longer works for non-ideal EOS wherein u can be less than 0.  Any thoughts here?

Reply:
I suggest that we floor the energy/temperature inside the solver, as it's only used for guesses

Reply:
> I unresolved this conversation because I was concerned that `PTESolverRhoU ` now no longer works for non-ideal EOS wherein u can be less than 0. Any thoughts here?

If `PTESolverRhoU` relies on a negative temperature guess, I'd argue that it never worked for non-ideal EOS that can have negative energies and that the guess infrastructure should be re-worked for that solver.

Reply:
Ah I misread the above. I forgot that `TryIdealPTE()` is separate from `GetIdealPTE()`. So these error checks being located in `GetIdealPTE()` is a mistake since `TryIdealPTE()` will check the results of `GetIdealPTE()` and see if the results are sane.

Even though these changes are highlighted just because I changed the message and not because they were added here, I'll move them to the `TryIdealPTE()` function.

Reply:
Actually, I ended up getting rid of all the `GetIdealPTE` asserts and instead I added a check in `TryIdealPTE()` to make sure the temperature guess is actually positive.